clang/lib/Sema/SemaDeclCXX.cpp

```diff
@@ -5450,10 +5450,20 @@ bool Sema::SetCtorInitializers(CXXConstructorDecl *Constructor, bool AnyErrors,
            NumInitializers * sizeof(CXXCtorInitializer*));
     Constructor->setCtorInitializers(baseOrMemberInitializers);
 
+    SourceLocation Location = Constructor->getLocation();
+
     // Constructors implicitly reference the base and member
     // destructors.
-    MarkBaseAndMemberDestructorsReferenced(Constructor->getLocation(),
-                                           Constructor->getParent());
+
+    for (CXXCtorInitializer *Initializer : Info.AllToInit) {
+      FieldDecl *Field = Initializer->getAnyMember();
+      if (!Field)
+        continue;
+
+      MarkFieldDestructorReferenced(Location, Field);
```

Comment:
We should include a standard reference here, I believe the correct reference would be: https://eel.is/c++draft/class.base.init#12

Reply:
done